includes/Story_Post_Type.php

```diff
@@ -252,6 +254,79 @@ public function register_meta(): void {
 		);
 	}
 
+	/**
+	 * Filters whether a post has a post thumbnail.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param bool             $has_thumbnail true if the post has a post thumbnail, otherwise false.
+	 * @param int|WP_Post|null $post          Post ID or WP_Post object. Default is global `$post`.
+	 */
+	public function has_post_thumbnail( $has_thumbnail, $post ): bool {
+		$post = get_post( $post );
+		if ( $post && ! $has_thumbnail && $this->get_slug() === get_post_type( $post ) ) {
+			$poster        = get_post_meta( $post->ID, self::POSTER_META_KEY, true );
+			$has_thumbnail = (bool) $poster;
+		}
+
+		return $has_thumbnail;
+	}
+
+	/**
+	 * Filters the post thumbnail HTML to return a img of hotlink image.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param string                       $html              The post thumbnail HTML.
+	 * @param int                          $post_id           The post ID.
+	 * @param int                          $post_thumbnail_id The post thumbnail ID, or 0 if there isn't one.
+	 * @param string|int[]                 $size              Requested image size. Can be any registered image size name, or
+	 *                                                        an array of width and height values in pixels (in that order).
+	 * @param string|array<string, string> $attr              Query string or array of attributes.
+	 * @return string The post thumbnail image tag.
+	 */
+	public function post_thumbnail_html( $html, $post_id, $post_thumbnail_id, $size, $attr ): string {
+		$post = get_post( $post_id );
+		if ( ! $post || $html || $post_thumbnail_id || $this->get_slug() !== get_post_type( $post ) ) {
```

Comment:
Is there a case is where the html could be empty but there's a  $post_thumbnail_id ? --- I haven't look at the code that consumes this --- could that cause issues?

Reply:
`$html` can be filtered, so it could be an empty string or something.